### PR TITLE
Set up more strict code styles and fix existing issues

### PR DESCRIPTION
**Pull request by kezhenxu94**
### Motivation:

Review code styles with some bots automatically.

### Modifications:

- Set up ReviewDog in GitHub Action to review code style.

- Add more check rules to checkstyle plugin.

### Result:

- Obvious code styles can be reviewed and commented automatically.

### CODE REVIEW

File: .github/workflows/ci-it.yaml [hunk 1]
```diff
@@ -38,11 +38,26 @@ jobs:
...
+          echo ::add-path::$HOME/bin
+          echo ::add-path::$(go env GOPATH)/bin # for Go projects
+
       - name: 'Install & Test'
         run: |
           export MAVEN_OPTS='-XX:+TieredCompilation -XX:TieredStopAtLevel=1 -XX:+CMSClassUnloadingEnabled -XX:+UseConcMarkSweepGC -XX:-UseGCOverheadLimit -Xmx3g'
-          ./mvnw --batch-mode -P"agent,backend,ui,dist,CI-with-IT" clean cobertura:cobertura verify install javadoc:javadoc
-          CODECOV_TOKEN="d2065307-8f01-4637-9715-2781ef096db7" bash <(curl -s https://codecov.io/bash)
+          ./mvnw --batch-mode -DskipTests -Drat.skip install # -P"agent,backend,ui,dist,CI-with-IT" clean cobertura:cobertura verify install javadoc:javadoc
+          # CODECOV_TOKEN="d2065307-8f01-4637-9715-2781ef096db7" bash <(curl -s https://codecov.io/bash)
```
Comment (wu-sheng) [inline .github/workflows/ci-it.yaml]: Why include `#`?

File: .github/workflows/ci-it.yaml [hunk 2]
```diff
@@ -38,11 +38,26 @@ jobs:
       - uses: actions/setup-java@v1
         with:
           java-version: 8
+
+      - name: Setup reviewdog
```
Comment (wu-sheng) [inline .github/workflows/ci-it.yaml]: Have you requested open `reviewdog` on GitHub?

File: .github/workflows/ci-it.yaml [hunk 3]
```diff
@@ -38,11 +38,26 @@ jobs:
...
         run: |
           export MAVEN_OPTS='-XX:+TieredCompilation -XX:TieredStopAtLevel=1 -XX:+CMSClassUnloadingEnabled -XX:+UseConcMarkSweepGC -XX:-UseGCOverheadLimit -Xmx3g'
-          ./mvnw --batch-mode -P"agent,backend,ui,dist,CI-with-IT" clean cobertura:cobertura verify install javadoc:javadoc
-          CODECOV_TOKEN="d2065307-8f01-4637-9715-2781ef096db7" bash <(curl -s https://codecov.io/bash)
+          ./mvnw --batch-mode -DskipTests -Drat.skip install # -P"agent,backend,ui,dist,CI-with-IT" clean cobertura:cobertura verify install javadoc:javadoc
+          # CODECOV_TOKEN="d2065307-8f01-4637-9715-2781ef096db7" bash <(curl -s https://codecov.io/bash)
+
+      - name: Run reviewdog
+        env:
+          REVIEWDOG_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (wu-sheng) [inline .github/workflows/ci-it.yaml]: Do we have this token?

File: .github/workflows/ci-it.yaml (hunk 1 above)
Comment (kezhenxu94) [reply .github/workflows/ci-it.yaml]: Obviously just for quick testing, will revert when test finished

File: .github/workflows/ci-it.yaml (hunk 2 above)
Comment (kezhenxu94) [reply .github/workflows/ci-it.yaml]: no need

File: .github/workflows/ci-it.yaml (hunk 3 above)
Comment (kezhenxu94) [reply .github/workflows/ci-it.yaml]: It's built in GitHub Action

**Comment by kezhenxu94** [comment]
Tests result look good

**Comment by kezhenxu94** [comment]
I've just added checks for `@author` tag and empty JavaDoc tags (`@param`, `@return`, etc.), which is the most obvious problems that we're facing from the current contributors, if I miss some, please let me know

File: .github/workflows/ci-it.yaml [hunk 4]
```diff
@@ -38,11 +38,26 @@ jobs:
       - uses: actions/setup-java@v1
         with:
           java-version: 8
+
+      - name: Setup reviewdog
+        run: |
+          mkdir -p $HOME/bin && curl -sfL https://raw.githubusercontent.com/reviewdog/reviewdog/master/install.sh| sh -s -- -b $HOME/bin
+          echo ::add-path::$HOME/bin
+          echo ::add-path::$(go env GOPATH)/bin # for Go projects
```
Comment (wu-sheng) [inline .github/workflows/ci-it.yaml]: Are these copy/paste issue?

File: .github/workflows/ci-it.yaml [hunk 5]
```diff
@@ -38,11 +38,26 @@ jobs:
...
         run: |
           export MAVEN_OPTS='-XX:+TieredCompilation -XX:TieredStopAtLevel=1 -XX:+CMSClassUnloadingEnabled -XX:+UseConcMarkSweepGC -XX:-UseGCOverheadLimit -Xmx3g'
           ./mvnw --batch-mode -P"agent,backend,ui,dist,CI-with-IT" clean cobertura:cobertura verify install javadoc:javadoc
           CODECOV_TOKEN="d2065307-8f01-4637-9715-2781ef096db7" bash <(curl -s https://codecov.io/bash)
+
+      - name: Run reviewdog
+        env:
+          REVIEWDOG_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          ./mvnw clean -Dcheckstyle.fails.on.error=false checkstyle:checkstyle-aggregate
```
Comment (wu-sheng) [inline .github/workflows/ci-it.yaml]: If this gets any chance to run? If checkstyle fail, the `Install & Test` step should stop the CI, no?

File: apm-checkstyle/checkStyle.xml [hunk 6]
```diff
@@ -20,19 +20,27 @@
...
+        <property name="eachLine" value="true"/>
+    </module>
 
     <module name="RegexpSingleline">
         <property name="format" value="System\.out\.println"/>
         <property name="message" value="Prohibit invoking System.out.println in source code !"/>
     </module>
 
+    <module name="RegexpSingleline">
+        <property name="severity" value="warning"/>
```
Comment (wu-sheng) [inline apm-checkstyle/checkStyle.xml]: Why this is warning? I think we should remove them? (Existing @author)

File: apm-checkstyle/checkStyle.xml [hunk 7]
```diff
@@ -123,4 +124,21 @@
         <property name="multiLines" value="1, 2, 3, 18"/>
     </module>
 
+    <module name="TreeWalker">
+        <property name="severity" value="warning"/>
```
Comment (wu-sheng) [inline apm-checkstyle/checkStyle.xml]: Same question here. Why warning?

**Comment by wu-sheng** [comment]
> I've just added checks for @author tag and empty JavaDoc tags (@param, @return, etc.), which are the most obvious problems that we're facing from the current contributors, if I miss some, please let me know

I think you should set them as an error, as removing the illegal ones once for all.

**Comment by wu-sheng** [comment]
Does this review only cover the changes? I remember there are plenty of `author`, it didn't indicate them.

**Comment by wu-sheng** [comment]
Related to #4272

**Comment by wu-sheng** [comment]
Please fix the conflicts. Besides the unstable profile e2e(will be fixed soon), there are other tests failing, please recheck.

**Comment by wu-sheng** [comment]
Seems still failing. :P

**Comment by wu-sheng** [comment]
2790 files changed. :P How this could be review.  Haha.

**Comment by kezhenxu94** [comment]
> 2790 files changed. :P How this could be review. Haha.

I can't even review myself 😢 , that's why I configure an incremental check before https://github.com/apache/skywalking/pull/4337#issuecomment-583848327

FYI, I just configure the rules and hit `Command+Option+L`, use regular expression to remove `@author`s, they effect mostly in just JavaDoc and `import`s, so as long as the tests passed, it should be generally OK then

**Comment by wu-sheng** [comment]
Don't worry about profile e2e, it has a code issue which has been fixed in #4335. Waiting CI confirmation to merge it.

**Comment by wu-sheng** [comment]
Merged, and conflicts showing up. Please fix, let's make this PR merged as next to avoid endless conflicts.

**Comment by wu-sheng** [comment]
Fail in spring case every time, strange though. :P

**Comment by wu-sheng** [comment]
Is the agent compiling test failure from one to another because of network issue? The CI passed, so I assume the compiling should be fine.

**Comment by wu-sheng** [comment]
> [ERROR] Failed to execute goal on project feign-scenario: Could not resolve dependencies for project org.apache.skywalking.apm.testcase:feign-scenario:jar:1.0.0: Could not transfer artifact io.github.openfeign:feign-core:jar:9.2.0 from/to central (https://repo.maven.apache.org/maven2): Failed to transfer file https://repo.maven.apache.org/maven2/io/github/openfeign/feign-core/9.2.0/feign-core-9.2.0.jar with status code 500 -> [Help 1]

Seems like the Maven central is not stable.

**Comment by wu-sheng** [comment]
Rechecked from locally, https://repo.maven.apache.org/maven2/io/github/openfeign/feign-core/9.2.0/feign-core-9.2.0.jar download is good.

**Comment by wu-sheng** [comment]
@kezhenxu94 Do you agree we should merge this than waiting the CI fails again and again because of network issue?

**Comment by kezhenxu94** [comment]
@apache/skywalking-committers please update the codes locally and reimport the `codeStyle.xml` into your IDEA, thanks :)

**Comment by wu-sheng** [comment]
@kezhenxu94 Send a notification mail to dev ml please.

**Pull request merged**